lib/stdlib/src/supervisor.erl

```diff
@@ -334,35 +321,34 @@ init_dynamic(_State, StartSpec) ->
 
 %%-----------------------------------------------------------------
 %% Func: start_children/2
-%% Args: Children = [child_rec()] in start order
+%% Args: Children = children() % Ids in start order
 %%       SupName = {local, atom()} | {global, atom()} | {pid(), Mod}
-%% Purpose: Start all children.  The new list contains #child's
+%% Purpose: Start all children.  The new map contains #child's
 %%          with pids.
 %% Returns: {ok, NChildren} | {error, NChildren, Reason}
-%%          NChildren = [child_rec()] in termination order (reversed
-%%                        start order)
+%%          NChildren = children() % Ids in termination order
+%%                                   (reversed start order)
 %%-----------------------------------------------------------------
-start_children(Children, SupName) -> start_children(Children, [], SupName).
-
-start_children([Child|Chs], NChildren, SupName) ->
-    case do_start_child(SupName, Child) of
-	{ok, undefined} when Child#child.restart_type =:= temporary ->
-	    start_children(Chs, NChildren, SupName);
-	{ok, Pid} ->
-	    start_children(Chs, [Child#child{pid = Pid}|NChildren], SupName);
-	{ok, Pid, _Extra} ->
-	    start_children(Chs, [Child#child{pid = Pid}|NChildren], SupName);
-	{error, Reason} ->
-	    report_error(start_error, Reason, Child, SupName),
-	    {error, lists:reverse(Chs) ++ [Child | NChildren],
-	     {failed_to_start_child,Child#child.name,Reason}}
-    end;
-start_children([], NChildren, _SupName) ->
-    {ok, NChildren}.
+start_children(Children, SupName) ->
+    Start =
```

Comment:
Is there a good reason why this function is inline instead of being its own named function?  It would help make stacktraces more clear, imo, if the `Start` function was named something like `attempt_start_child/2` or something.

Reply:
No, there is no particular reason for this. I'll make it a separate function. Thanks!

Reply:
Sorry, I answered this one a bit too quickly. The reason for it being inline is that it uses the bound variable SupName. Moving it out would make the code less clean, and I don't believe that it is really justified by the gain. As is, the stack trace of a crashed child start function is as follows:
```
** exception exit: {shutdown,
                       {failed_to_start_child,1,
                           {'EXIT',
                               {undef,
                                   [{test_gen_server,start_link,[],[]},
                                    {supervisor,do_start_child_i,3,[{file,"supervisor.erl"},{line,365}]},
                                    {supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,351}]},
                                    {supervisor,'-start_children/2-fun-0-',3,[{file,"supervisor.erl"},{line,335}]},
                                    {supervisor,ch_map,4,[{file,"supervisor.erl"},{line,1148}]},
                                    {supervisor,init_children,2,[{file,"supervisor.erl"},{line,301}]},
                                    {gen_server,init_it,2,[{file,"gen_server.erl"},{line,369}]},
                                    {gen_server,init_it,6,[{file,"gen_server.erl"},{line,337}]}]}}}}
```
which I believe is clear enough.